perf/src/recycler.rs

```diff
@@ -375,124 +171,26 @@ impl<T: Default + Reset> RecyclerX<T> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::packet::PacketsRecycler;
-    use std::{thread::sleep, time::Duration};
 
     impl Reset for u64 {
         fn reset(&mut self) {
             *self = 10;
         }
         fn warm(&mut self, _size_hint: usize) {}
         fn set_recycler(&mut self, _recycler: Weak<RecyclerX<Self>>) {}
-        fn unset_recycler(&mut self) {}
     }
 
     #[test]
     fn test_recycler() {
-        let recycler = Recycler::new_without_limit("");
-        let mut y: u64 = recycler.allocate().unwrap();
+        let recycler = Recycler::default();
+        let mut y: u64 = recycler.allocate("test_recycler1");
         assert_eq!(y, 0);
         y = 20;
         let recycler2 = recycler.clone();
         recycler2.recycler.recycle(y);
         assert_eq!(recycler.recycler.gc.lock().unwrap().len(), 1);
-        let z = recycler.allocate().unwrap();
+        let z = recycler.allocate("test_recycler2");
         assert_eq!(z, 10);
         assert_eq!(recycler.recycler.gc.lock().unwrap().len(), 0);
     }
-
-    #[test]
-    fn test_recycler_limit() {
-        let limit = 10;
-        assert!(limit <= DEFAULT_MINIMUM_OBJECT_COUNT);
-        // Use PacketRecycler so that dropping the allocated object
-        // actually recycles
-        let recycler = PacketsRecycler::new_with_limit("", limit);
-        let mut allocated_items = vec![];
-        for i in 0..limit * 2 {
-            let x = recycler.allocate();
-            if i < limit {
-                allocated_items.push(x.unwrap());
-            } else {
-                assert!(x.is_none());
-            }
-        }
-        assert_eq!(
-            recycler.recycler.gc.lock().unwrap().total_allocated_count,
-            limit
-        );
-        assert_eq!(recycler.recycler.gc.lock().unwrap().len(), 0_usize);
-        drop(allocated_items);
-        assert_eq!(
-            recycler.recycler.gc.lock().unwrap().total_allocated_count,
-            limit
-        );
-        assert_eq!(recycler.recycler.gc.lock().unwrap().len(), limit as usize);
-    }
-
-    #[test]
-    fn test_recycler_shrink() {
```

Comment:
Can we add an equivalent test for the shrinking path?

Reply:
added  a test coverage